### PR TITLE
Re-enable ADAM for 1D (non-angular) analysis

### PR DESCRIPTION
**Pull request by joglekara**
See https://continuum.ergodic.io/experiments/#/experiments/298/runs/4f51caccbed94c699ea9eeb1437771ba vs https://continuum.ergodic.io/experiments/#/experiments/298/runs/231ca63bf4714e71a56921615d11c6fb

### CODE REVIEW

**Comment by joglekara** [comment]
The main difference between ADAM (from JAXopt) and L-BFGS-B (from SciPy) is that the latter has an auto-convergence stop and often terminates fairly early (for better or worse). The former requires us to implement that.

Right now, it just runs to the maximum number of steps specified in the config and returns the parameters from the step with the best loss. It will be useful to implement an early stopping condition

It will then be useful to unify the codepath for 1D lineouts and branch where needed (optimizer in this case).

Then it will be useful to extend this to angular

**Comment by joglekara** [comment]
Also did some refactoring with the file storage

File: configs/1d/defaults.yaml
Comment (almilder) [inline configs/1d/defaults.yaml:1]: I am not seeing the sequential field in the test runs, does this work with adam?

Comment (almilder) [reply configs/1d/defaults.yaml:1]: NM i see how this works in the fitter function

File: run_tsadar.py
Comment (almilder) [inline run_tsadar.py:1]: please re-enable jit

**Review by almilder — CHANGES_REQUESTED** [review]
Generally looks good, but i think we need jit enabled

**Review by almilder — APPROVED** [review]

**Pull request merged**